src/index.js

```diff
@@ -307,51 +264,43 @@ class Bitswap {
    * @param {function(Error)} callback
    * @returns {void}
    */
-  put (block, callback) {
-    this._log('putting block')
-
-    waterfall([
-      (cb) => this.blockstore.has(block.cid, cb),
-      (has, cb) => {
-        if (has) {
-          return nextTick(cb)
-        }
+  async put (block) { // eslint-disable-line require-await
+    if (!Array.isArray(block)) {
+      block = [
+        block
+      ]
+    }
 
-        this._putBlock(block, cb)
-      }
-    ], callback)
+    return this.putMany(block)
   }
 
   /**
    * Put the given blocks to the underlying blockstore and
    * send it to nodes that have it them their wantlist.
    *
-   * @param {Array<Block>} blocks
+   * @param {AsyncIterable<Block>} blocks
```

Comment:
```suggestion
   * @param {AsyncIterable<Block>|Iterable<Block>} blocks
```

Reply:
Comments are not a type system. 😉

Reply:
they can be (ts can check types with jsdoc) and editor autocomplete is underrated (for me is very helpful)